astro/src/content/docs/lifecycle/migrate-users/bulk/pingone.mdx

```diff
@@ -0,0 +1,849 @@
+- You can use an AWS [Lambda function](https://docs.aws.amazon.com/lambda/latest/dg/welcome.html).
+- You can create a Flask app that exposes an endpoint used with ngrok and configured as a Connector on FusionAuth.
+
+This guide will show you how to use Flask and ngrok.
+
+The authentication stub accepts a login request from FusionAuth over TLS and performs an [ROPC login](https://docs.pingidentity.com/r/en-us/developer/kwc1601508084434) request on Ping Identity, returning the relevant user data via the PingOne API. 
+
+Create an OIDC application in your Ping Environment to call from the Flask app by following this [Ping One guide](https://docs.pingidentity.com/r/en-us/pingfederate-pingone-mfa-ik/pingfederate_pingone_mfa_ik_creating_a_web_or_native_application_in_pingone). 
+
+- Select "Code" as the "Response Type".
+- Take note of the Client Id of the application to use in the Flask app.
+
+<Aside type="caution">
+  FusionAuth Connectors will send user credentials to Ping Identity via a Python Flask app for demonstration purposes. In production migrations, it is important to secure both the data in transit and the various endpoints that form part of the solution.
+
+  In this guide, all communication is over TLS, and the Python Flask app can use a unique query string code for security purposes. Regardless of the migration implementation you choose, security in production migrations should be the highest priority and proper review and planning regarding this should be done.
+
+  Ping Identity also uses an access token that needs certain administrative permissions to perform the functions needed. Please bear in mind that the access token may expire and need to be renewed.
+</Aside>
+
+### Set Up Authentication Logic
+
+Create a `pingIdentityAuth.py` file and add the Flask authenticator code below to it.
+
+<RemoteCode lang="python" url={codeRoot + "/pingidentity/export/slowmigration/pingIdentityAuth.py"} />
+
+Modify the example function code provided as needed and remember to update the domains (for example, `apiPath`) to match the region where the Ping Identity data is hosted.
+
+This code receives a login request from FusionAuth, attempts to log the user in via ROPC, and then returns a FusionAuth user object on success.
+
+Following successful authentication, the function calls the [PingOne API `users` endpoint](https://apidocs.pingidentity.com/pingone/main/v1/api/#getting-started-with-the-pingone-apis) to get additional user attributes, which are then transformed into a FusionAuth-compatible format.
+
+Make sure that all requirements (like `requests` and `flask`) are installed with pip before you run the script.
+
+Now run the application.
+
+```sh
+python pingIdentityAuth.py
+```
+
+#### Testing The Deployed API
+
+The `transform_json_user` function converts the JSON returned from Ping Identity to the format FusionAuth requires.
```

Comment:
Need more details on how the application in PingOne should be setup. Including how to create `access token` if necessary. I added a comment on the script as well regarding this.